test_values/test_cases.json

```diff
@@ -252,5 +252,196 @@
       ],
       "expected": true
     }
+
+  ],
+  "deletion_tests": [
+    {
+      "leaf_values": [0, 1, 2, 3, 4, 5, 6, 7],
+      "target_values": [1, 7],
+      "roots": [
+        "332c306188d35eb22ecb05d8c00446cd6a7a475f6615f46207cfaa713bb3e62c"
+      ],
+      "proof_hashes": [
+        "6e340b9cffb37a989ca544e6bb780a2c78901d3fb33738768511a30617afa01d",
+        "67586e98fad27da0b9968bc039a1ef34c939b9b8e523a8bef89d478608c5ecf6",
+        "9576f4ade6e9bc3a6458b506ce3e4e890df29cb14cb5d3d887672aef55647a2b",
+        "9eec588c41d87b16b0ee226cb38da3864f9537632321d8be855a73d5616dcc73"
+      ]
+    },
+    {
+      "leaf_values": [0, 1, 2, 3, 4, 5, 6, 7],
+      "target_values": [1, 5, 7],
+      "roots": [
+        "3b8b6eb231437092f6d9fbf8a0696b7cb446e40f1bf81ddb23d3eabb3080b0dd"
+      ],
+      "proof_hashes": [
+       "6e340b9cffb37a989ca544e6bb780a2c78901d3fb33738768511a30617afa01d",
+       "e52d9c508c502347344d8c07ad91cbd6068afc75ff6292f062a09ca381c89e71",
+       "67586e98fad27da0b9968bc039a1ef34c939b9b8e523a8bef89d478608c5ecf6",
+       "9576f4ade6e9bc3a6458b506ce3e4e890df29cb14cb5d3d887672aef55647a2b"
+      ]
+    },
+    {
+      "leaf_values": [
```

Comment:
Nit: leaf_values for the test case below have different formatting but they're the same thing. Having the same formatting would be good